tests/sync/session.cpp

```diff
@@ -38,14 +38,18 @@ using namespace realm::util;
 template <typename FetchAccessToken, typename ErrorHandler>
 std::shared_ptr<SyncSession> sync_session(SyncServer& server, std::shared_ptr<SyncUser> user, const std::string& path,
                                           FetchAccessToken&& fetch_access_token, ErrorHandler&& error_handler,
-                                          SyncSessionStopPolicy stop_policy=SyncSessionStopPolicy::AfterChangesUploaded)
+                                          SyncSessionStopPolicy stop_policy=SyncSessionStopPolicy::AfterChangesUploaded,
+                                          std::string** on_disk_path=nullptr)
```

Comment:
This can just be a `std::string *`. There's no need for the double indirection.

An alternative to an output parameter if most callers were interested in both outputs would be to return a `std::pair` or `std::tuple`. Since this is rarely used I think the output parameter is ok.